src/odemis/acq/fastem.py

```diff
@@ -346,23 +289,22 @@ def run(self):
         :raise:
             Exception: If it failed before any single field images were acquired or if acquisition was cancelled.
         """
-        self._beamshift.shift.value = (0, 0)
-
         # Update the position of the first tile.
         self._pos_first_tile = self.get_pos_first_tile()
 
-        # Move the stage to the first tile, to ensure the autofocus and image
-        # translation pre-align are done in the correct location.
+        # Move the stage to the first tile, to ensure the autofocus and image translation pre-align are
+        # done in the correct location and the correct position is stored in the megafield metadata yaml file.
         self.move_stage_to_next_tile()
 
-        exception = self.pre_calibrate()
-
-        if exception is not None:
-            return None, exception
+        # If we run the pre-calibrate check those exceptions, else initialize exceptions as None.
```

Comment:
I am confused by the comment. Where do you check the exceptions? You catch exception in the self.pre_calibrate(). If an exception is raised, it is handled by the caller right?
The exception for the ROA acquisition is always initialized with None. Independent of the pre-calibrations. I suggest to just remove the comment completely unless I misunderstood what you wanted to say.

Reply:
This was indeed left over from a previous commit